src/app/test/matchers/toHaveComputedStyle.ts

```diff
@@ -0,0 +1,76 @@
+import CustomMatcherFactories = jasmine.CustomMatcherFactories;
+import CustomMatcher = jasmine.CustomMatcher;
+import CustomMatcherResult = jasmine.CustomMatcherResult;
+
+/**
+ * Custom matchers use a combination of the following resources:
+ * https://stackoverflow.com/questions/42956195/create-custom-jasmine-matcher-using-typescript
+ * https://jasmine.github.io/tutorials/custom_matcher
+ *
+ * Remember to update matcher-types.d.ts when making changes
+ */
+
+type StyleValue = string | number;
+
+function validateStyles(
+  actual: HTMLElement,
+  expected: Partial<CSSStyleDeclaration>,
+  callback: (
+    key: string,
+    actualStyle: StyleValue,
+    expectedStyle: StyleValue
+  ) => CustomMatcherResult
+): CustomMatcherResult {
+  const computedStyle = getComputedStyle(actual);
```

Comment:
silly question: where does this function become available? Is it not imported? Are we pulling it off a global scope?

Reply:
Made available here: https://github.com/QutEcoacoustics/workbench-client/blob/bdeeb1863ff4ca2e2efd373238baa76be92596c1/src/test.ts#L17-L21